libsolidity/interface/CompilerStack.cpp

```diff
@@ -1251,14 +1264,47 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	tie(compiledContract.yulIR, compiledContract.yulIROptimized) = generator.run(_contract, otherYulSources);
 }
 
+void CompilerStack::generateEVM(ContractDefinition const& _contract)
+{
+	solAssert(m_stackState >= AnalysisPerformed, "");
+	if (m_hasError)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateEVM with errors."));
+
+	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
+	// Empty contract, ignore.
+	// TODO: check if this could happen in other cases.
+	if (compiledContract.yulIROptimized.empty())
+		return;
+	if (!compiledContract.object.bytecode.empty())
+		return;
+
+	// Re-parse the Yul IR in EVM dialect
+	yul::AssemblyStack stack(m_evmVersion, yul::AssemblyStack::Language::StrictAssembly, m_optimiserSettings);
+	stack.parseAndAnalyze("", compiledContract.yulIROptimized);
+	stack.optimize();
+
+	//cout << yul::AsmPrinter{}(*stack.parserResult()->code) << endl;
+
+	// TODO: support passing metadata
+	auto result = stack.assemble(yul::AssemblyStack::Machine::EVM);
```

Comment:
Apparently could use `stack.assembleAndGuessRuntime()` which outputs the runtime object too, but that function looks like such a "hack" :)

Reply:
`IRNames::runtimeObject(_contract)`

Reply:
You mean to extract the object from the `stack` using that name? We do not seem to have a feature to access subobjects from within the `stack`.

Reply:
If we go without, maybe create an issue to keep track?

Reply:
There is an issue: #10152.